pkg/controlplane/instance.go

```diff
@@ -754,3 +787,30 @@ func DefaultAPIResourceConfigSource() *serverstorage.ResourceConfig {
 
 	return ret
 }
+
+// Utility function to check if the UnknownVersionInteroperabilityProxy feature flag is enabled.
+// Since UnknownVersionInteroperabilityProxy feature depends on APIServerIdentity and StorageVersionAPI
+// features, this function checks if all the three flags are enabled.
+func PeerProxyFeatureEnabled() bool {
```

Comment:
same comments dropping logging from this helper method, validating the features are set coherently at startup, and probably dropping this helper method entirely in favor of a single `utilfeature.DefaultFeatureGate.Enabled(apiserverfeatures.UnknownVersionInteroperabilityProxy)` call

Reply:
removed this method.